bundle/src/main/resources/gradle/dependencies.gradle

```diff
@@ -0,0 +1,77 @@
+allprojects {
+	task generateDependenciesGraphAsJson() {
+		doLast {
+			Set<Configuration> configurationSet = project.getConfigurations();
+			String jsonOutput = ""
+			jsonOutput += "{\"groupId\":\"" + project.getGroup() + "\",\"artifactId\":\"" + project.getName() + "\",\"version\":\"" +  project.getVersion() + "\""
+			jsonOutput += ",\"dependencies\":["
+			Spec<Object> SATISFIES_ALL = new Spec<Object>() {
+				public boolean isSatisfiedBy(Object element) {
+					return true;
+				}
+			};
+		
+			for (Configuration configuration : configurationSet) {
+				try {
+					ResolvedConfiguration resolvedConfiguration = configuration.getResolvedConfiguration();
+					LenientConfiguration lenientConfiguration = resolvedConfiguration.getLenientConfiguration();
+					Set<ResolvedDependency> firstLevelModuleDependencies = lenientConfiguration.getFirstLevelModuleDependencies(SATISFIES_ALL);
+					Iterator<ResolvedDependency> iterator = firstLevelModuleDependencies.iterator();
+					while (iterator.hasNext()) {
+						ResolvedDependency resolvedDependency = iterator.next();
+						jsonOutput += "{\"groupId\":\"" + resolvedDependency.getModuleGroup() + "\",\"artifactId\":\"" + resolvedDependency.getModuleName() + "\",\"version\":\"" + resolvedDependency.getModuleVersion() + "\""
+						Set<ResolvedDependency> childrens = resolvedDependency.getChildren();
+						jsonOutput += ",\"dependencies\":["
+						jsonOutput = printChildrens(childrens, jsonOutput);
+						jsonOutput += "]\n"
+						jsonOutput += "},"
+					}
+					Set<UnresolvedDependency> unresolvedDependencies = lenientConfiguration.getUnresolvedModuleDependencies();
+					Iterator<UnresolvedDependency> iteratorUnresolvedDependency = unresolvedDependencies.iterator();
+					while (iteratorUnresolvedDependency.hasNext()) {
+						UnresolvedDependency unresolvedDependency = iteratorUnresolvedDependency.next();
+						ModuleVersionSelector selector = unresolvedDependency.getSelector();
+						jsonOutput += "{\"groupId\":\"" + selector.getGroup() + "\",\"artifactId\":\"" + selector.getName() + "\",\"version\":\"" +  selector.getVersion() + "\""
+						jsonOutput += ",\"dependencies\":["
+						jsonOutput += "]\n"
+						jsonOutput += "},"
+					}
+				} catch (RuntimeException e) {
+					// ignore
+				}
+			}
+			if (jsonOutput.endsWith(",")) {
+				jsonOutput = jsonOutput.substring(0, jsonOutput.length() - 1)
+			}
+			jsonOutput += "]}"
+			if (jsonOutput.equals("[]")) {
+				return
+			}
+			File homeDir = new File(System.getProperty("user.home") + File.separator + "jfrog-eclipse-plugin" + File.separator + name + File.separator + project.getName());
+			if (!homeDir.exists()) {
+				homeDir.mkdirs();
+			}
+
+			BufferedWriter writer = new BufferedWriter(new FileWriter(homeDir.getAbsolutePath() + File.separator + project.getName() + ".txt"));
```

Comment:
1. FileWriter also need to be closed.
2. Please use `try...finally` or `try-with-resources` to close resources.
3. Please use Paths.get(homeDir.getAbsolutePath(), project.getName() + ".txt")

Reply:
Groovy is not supporting the java try with resources, Ive added try finally.. I've also removed the BufferedWriter.